hlib/python/hlib/frontend/test/test_elemwise.py

```diff
@@ -0,0 +1,225 @@
+assert_mul(*elemwise_test((3,), hlib.op.op.elemwise_mul, hcl.Int()))
+assert_mul(*elemwise_test((3, 3, 3, 3, 3), hlib.op.op.elemwise_mul, hcl.Int()))
+
+assert_mul(*elemwise_test((3, 3), hlib.op.op.elemwise_mul, hcl.Float()))
+assert_mul(*elemwise_test((3, 3, 3), hlib.op.op.elemwise_mul, hcl.Float()))
+assert_mul(*elemwise_test((3,), hlib.op.op.elemwise_mul, hcl.Float()))
+assert_mul(
+    *
+    elemwise_test(
+        (3,
+         3,
+         3,
+         3,
+         3),
+        hlib.op.op.elemwise_mul,
+        hcl.Float()))
+
+assert_div_int(*elemwise_test((3, 3), hlib.op.op.elemwise_div, hcl.Int()))
+assert_div_int(*elemwise_test((3, 3, 3), hlib.op.op.elemwise_div, hcl.Int()))
+assert_div_int(*elemwise_test((3,), hlib.op.op.elemwise_div, hcl.Int()))
+assert_div_int(
+    *
+    elemwise_test(
+        (3,
+         3,
+         3,
+         3,
+         3),
+        hlib.op.op.elemwise_div,
+        hcl.Int()))
+
+assert_div_float(*elemwise_test((3, 3), hlib.op.op.elemwise_div, hcl.Float()))
+assert_div_float(
+    *
+    elemwise_test(
+        (3,
+         3,
+         3),
+        hlib.op.op.elemwise_div,
+        hcl.Float()))
+assert_div_float(*elemwise_test((3,), hlib.op.op.elemwise_div, hcl.Float()))
+assert_div_float(
+    *
+    elemwise_test(
+        (3,
+         3,
+         3,
+         3,
+         3),
+        hlib.op.op.elemwise_div,
+        hcl.Float()))
+
+assert_mod(*elemwise_test((3, 3), hlib.op.op.elemwise_mod, hcl.Int()))
+assert_mod(*elemwise_test((3, 3, 3), hlib.op.op.elemwise_mod, hcl.Int()))
+assert_mod(*elemwise_test((3,), hlib.op.op.elemwise_mod, hcl.Int()))
+assert_mod(*elemwise_test((3, 3, 3, 3, 3), hlib.op.op.elemwise_mod, hcl.Int()))
+
+assert_mod(*elemwise_test((3, 3), hlib.op.op.elemwise_mod, hcl.Float()))
+assert_mod(*elemwise_test((3, 3, 3), hlib.op.op.elemwise_mod, hcl.Float()))
+assert_mod(*elemwise_test((3,), hlib.op.op.elemwise_mod, hcl.Float()))
+assert_mod(
+    *
+    elemwise_test(
+        (3,
+         3,
+         3,
+         3,
+         3),
+        hlib.op.op.elemwise_mod,
+        hcl.Float()))
+
+#assert_pow(*elemint_test((3, 3), hlib.op.op.elemwise_pow))
+#assert_pow(*elemint_test((3, 3, 3), hlib.op.op.elemwise_pow))
+#assert_pow(*elemint_test((3,), hlib.op.op.elemwise_pow))
+#assert_pow(*elemint_test((3, 3, 3, 3, 3), hlib.op.op.elemwise_pow))
+
+#assert_and(*elemint_test((3, 3), hlib.op.op.logical_and))
+# assert_and(*elemint_test((3,3,3),hlib.logical_and))
+# assert_and(*elemint_test((3,),hlib.logical_and))
+# assert_and(*elemint_test((3,3,3,3,3),hlib.logical_and))
+
+# assert_or(*elemint_test((3,3),hlib.logical_or))
+# assert_or(*elemint_test((3,3,3),hlib.logical_or))
+# assert_or(*elemint_test((3,),hlib.logical_or))
+# assert_or(*elemint_test((3,3,3,3,3),hlib.logical_or))
+
+# assert_not(*elemnot_test((3,3)))
+# assert_not(*elemnot_test((3,3,3)))
+# assert_not(*elemnot_test((3,)))
+# assert_not(*elemnot_test((3,3,3,3,3)))
```

Comment:
So are these tested?

Reply:
pow seems to be having issues for some reason and assert_not is throwing another issue, but that currently is not being used.